src/helpers/disclosure.ts

```diff
@@ -26,9 +26,17 @@ export function createDisclosure({
 	}
 
 	const [visible, setVisible] = createSignal(false)
+	const main = document.querySelector("main")
+	const footer = document.querySelector("footer")
 
 	createEffect(() => {
 		button.setAttribute("aria-expanded", String(visible()))
+		if (main) {
+			main.inert = visible()
+		}
+		if (footer) {
+			footer.inert = visible()
+		}
```

Comment:
Doing an if check on these is a silent failure if the elements happen to change 🫠

I'd consider something like this:
```ts
	const main = expect(document.querySelector("main"))
	const footer = expect(document.querySelector("footer"))
```
```ts
// in some helpers file
export function expect<T>(value: T): NonNullable<T> {
	if (value == null) {
		throw new Error(`Expected a non-nullable value, got ${value}`)
	}
	return value
}
```

Reply:
Also... it seems kind of scary for this to operate on global elements? It was created to mostly be a self-contained abstraction 🤔 

I also just realized that this is written in a way that `main` and `footer` might not exist on the page, so ignore the first comment I suppose! 🤪

Reply:
Ahh right, that's a good catch. This file is in the `src/helpers` folder, but the only place it is used is for the header... I agree that this global stuff shouldn't go here, though. I'll come up with something that makes more sense.

Reply:
Updated with an EventEmitter pattern in [394e38e](https://github.com/withastro/astro.build/pull/1033/commits/394e38ec4480a478dd3b6e54b0390452e8677e9d)